odoo/addons/base/models/ir_model.py

```diff
@@ -1319,6 +1319,86 @@ def get_field_selection(self, model_name, field_name):
         field = self._get(model_name, field_name)
         return [(sel.value, sel.name) for sel in field.selection_ids]
 
+
+class ModelInherit(models.Model):
+    _name = "ir.model.inherit"
+    _description = "Model Inheritance Tree"
+
+    model_id = fields.Many2one("ir.model", required=True, ondelete="cascade")
+    inherit_id = fields.Many2one("ir.model", required=True, ondelete="cascade")
+    field_id = fields.Many2one("ir.model.fields", ondelete="cascade")  # in case of inherits
+
+    _sql_constraints = [
+        ("uniq", "UNIQUE(model_id, inherit_id)", "")
+    ]
+
+    def _reflect_inherits(self, model_names):
+        """ Reflect the given models' inherits. """
+        get_model = self.env["ir.model"]._get_id
+        inherits = []
+
+        by_module = defaultdict(list)
+        for model_name in model_names:
+            get_field = self.env["ir.model.fields"]._get_ids(model_name).get
+            model_id = get_model(model_name)
+            model = self.env[model_name]
+
+            for cls in reversed(type(model).mro()):
+                if not models.is_definition_class(cls):
+                    continue
+
+                inherits = [
+                    (model_id, get_model(inherit), None) for inherit in cls._inherit
+                    if inherit != "base"
+                    if inherit != model_name
+                ]
+                inherits += [
+                    (model_id, get_model(inherit), get_field(field))
+                    for inherit, field in cls._inherits.items()
+                ]
+
+                for inh in inherits:
+                    by_module[inh].append(cls._module)
```

Comment:
The variable name `by_module` is totally confusing.  I would have expected a mapping from module name to a list of inherit items, but got it the other way around.  And BTW I don't see the point of the mapping at all.  A list/set with all the items (including the module) would do the job, wouldn't it?

Reply:
The variable can be renamed, indeed.

The mapping is needed to have the names of the modules that inherit those models to generate the xmlids.

Reply:
I see, so multiple _inherit(s) correspond to a single record with multiple external ids :+1: